src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -290,21 +387,96 @@ protected int getMaxAttacksUntilRestock()
     protected AIState patrol()
     {
         worker.setAIMoveSpeed(1);
+        final AbstractBuilding building = getOwnBuilding();
 
-        if (currentPatrolTarget == null)
+        if (building != null && building instanceof BuildingGuardTower)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) No need to check for null before an instanceof [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ASimplifyConditional)